services/case_filters/case_filter_base.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from abc import abstractmethod
+from typing import List
+
+from models.blaise.blaise_case_information_model import BlaiseCaseInformationModel
+from models.totalmobile.totalmobile_world_model import TotalmobileWorldModel
+
+
+class CaseFilterBase:
+    @property
+    @abstractmethod
+    def wave_number(self) -> int:
+        pass
+
+    @abstractmethod
+    def case_is_eligible(self, case: BlaiseCaseInformationModel) -> bool:
+        pass
+
+    @staticmethod
+    def case_is_part_of_wave(
+        wave_number: int, case: BlaiseCaseInformationModel
+    ) -> bool:
+        if case.wave == wave_number:
+            return True
+
+        return False
+
+    @staticmethod
+    def telephone_number_is_empty(case: BlaiseCaseInformationModel) -> bool:
+        if case.contact_details.telephone_number_1 == "":
+            return True
+
+        logging.info(
```

Comment:
Lot's of repetition with these log messages, possibly worked extracted in private method. It could even go a step further and be dynamically generated somehow.

Reply:
A simple versions might be to replace the body of this function with something like:

```python
 def telephone_number_is_empty(case: BlaiseCaseInformationModel) -> bool:
    return self._with_logging_on_failure(
        case.contact_details.telephone_number_1 == "",
        case,
        "telephone_number_1"
    )
```

Where

```python
    @staticmethod
    def _with_logging_on_failure(success: bool, case: BlaiseCaseInformationModel, field: str) -> bool:
        if success:
            return True

        logging.info(
            f"Case '{case.case_id}' in questionnaire '{case.questionnaire_name}' was not eligible to be sent to Totalmobile as it has a value set for the field '{field}'"
        )
        return False
````

Reply:
Its currently easy to follow, I hate the spurious logging and I dont think we should have it at all at this level. I think there should be one log which states its ineligible and output the case

Reply:
I would rather get rid of all the spurious logging and just log one ineligible case message with all properties